### PR TITLE
Hmis071 integrated antenatal encounter form

### PR DESCRIPTION
**Pull request by moshonk**
@ssmusoke, I have on the below for the AN C form. Please have a look at it
**1. Routine Administration** 
1. IPT Dose - add number 3 does 
2. TB Status – change Suspect to “Presumptive (Suspect)” 
3. TB Status – add the codes for TB from the register 
**2. Vitals** 
1. MUAC - Currently the MUAC codes are shown, however the register also requires the MUAC values to be entered so add a field for the MUAC values. 
2. MUAC – add a validation that checks the MUAC numeric value against the code – prompt the user but do not stop the form from saving. 
**3. Mother’s Information** 
1. Currently the dates are computed from the Last Normal Menstruation Period (LNNP) which is required, however the register has the gestation and Expected date of delivery. So change the computation as follows:
· Make the gestation and Expected Date of Delivery required fields 
· Use the EDD to compute the LNMP 
· ANC1 timing should be computed from the LNMP date 
· ANC timing should only be selected if this is the first ANC visit otherwise it should be disabled 
· Validate that the visit date falls between the EDD and LNMP  
2. EMTCT Code 
**4. Referral Information** 
1. Referral – add the descriptions of the codes as other drop downs 
2. Referral – Change the label to “Referral Type” 
Add a new text field “Referral Notes”
cc @ningosi

### CODE REVIEW

**Comment by moshonk** [comment]
@ssmusoke in the latest commit, I have worked on additional 2 items as below:
**HIV INFORMATION**
· Add a label that shows two values, Suppressed and Non-Suppressed after the viral load count. The suppressed value is selected for viral loads below 
Mother’s Information 
**EMTCT Code** 
. When a patient is HIV positive on the visit date, when the Summary page encounter is before the ANC encounter date, the EMTCT codes have to be updated as per the register  
cc @ningosi

**Comment by ssmusoke** [comment]
@moshonk Comments 

- LNMP is not a required field 
- IPT Does: for 1, 2 and 3 add explanations IPT 1, IPT2 and IPT 3, for C add Completed
- Mother's information: ANC timing not computed from LNMP but is computed from Gestational the which is not correct 
- ANC timing is not tied to only the first ANC visit 
- Validation for visit date between LNMP and EDD should read "The visit date must fall between the Expected Date of Delivery and L.N.M.P"

**Comment by moshonk** [comment]
@ssmusoke, Please find my comments below:
Some of your comments above seem to contradict the notes you shared
earlier. Please see below some of your comments above vis a vis screenshots
from the comments that you had shared.

LNMP is not a required field
[image: Inline images 1]
ANC timing not computed from LNMP but is computed from Gestational the
which is not correct
[image: Inline images 2]
ANC timing is not tied to only the first ANC visit
[image: Inline images 3]

@ningosi

Kindly provide clarification on the above.

Regards

Ken Omondi
Skype: omondi.ken84
Twitter: @moshonk
Yahoo: kenneth_omondi@yahoo.com

**Comment by moshonk** [comment]
@ssmusoke This is ready for review

**Comment by ssmusoke** [comment]
@moshonk The ANC 1 timing does not change when the ANC visit is 1 and the EDD is entered

**Comment by ssmusoke** [comment]
@moshonk This PR now has conflicts in the MCH concepts file

**Comment by ningosi** [comment]
@moshonk ressolve conflicts for this too and commit for review

File: omod/src/main/webapp/resources/htmlforms/071-IntegratedAntenatalRegister.xml [hunk 1]
```diff
@@ -682,8 +931,8 @@
                         <p>
                             <label>IPT Dose</label>
                             <obs conceptId="1da3cb98-59d8-4bfd-b0bb-c9c1bcd058c6"
-                                 answerConceptIds="0192ca59-b647-4f88-b07e-8fda991ba6d6,f1d5afce-8dfe-4d2d-b24b-051815d61848,f29f43c6-076a-4f70-9ae8-4563ac3fda80,dca06bae-30ab-102d-86b0-7a5022ba4115,fca28768-50dc-4d6b-a3d2-2aae3b376b27,b0439b8e-469e-43e6-9dda-9b6b49f2147b"
-                                 answerLabels="1,2,ND,C,CTX,CTX✔"
+                                 answerConceptIds="0192ca59-b647-4f88-b07e-8fda991ba6d6,f1d5afce-8dfe-4d2d-b24b-051815d61848,a5497b5a-7da1-42d2-9985-b5ec695b4199,f29f43c6-076a-4f70-9ae8-4563ac3fda80,dca06bae-30ab-102d-86b0-7a5022ba4115,fca28768-50dc-4d6b-a3d2-2aae3b376b27,b0439b8e-469e-43e6-9dda-9b6b49f2147b"
+                                 answerLabels="1 - IPT1,2 - IPT2,3 - IPT3,ND - Not Due,C - Complete,CTX - Initial  septrin prescrition,CTX✔ - Septrin prescribed on prev visit"
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/071-IntegratedAntenatalRegister.xml]: Change the tick to &#10003; as its breaking the forms when saving to different databases

**Comment by ssmusoke** [comment]
@moshonk Please change https://github.com/moshonk/openmrs-module-aijar/blob/f0bc3c7de3f618d0aaa0c8bfcc7a8dd8fe5552d8/omod/src/main/webapp/resources/htmlforms/071-IntegratedAntenatalRegister.xml#L842 to use HTML entities instead of the ticks which break the forms in certain cases

**Pull request merged**